modules/redis/redis.go

```diff
@@ -54,91 +90,97 @@ func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomize
 func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*RedisContainer, error) {
 	req := testcontainers.ContainerRequest{
 		Image:        img,
-		ExposedPorts: []string{"6379/tcp"},
-		WaitingFor:   wait.ForLog("* Ready to accept connections"),
+		ExposedPorts: []string{redisPort},
 	}
 
 	genericContainerReq := testcontainers.GenericContainerRequest{
 		ContainerRequest: req,
 		Started:          true,
 	}
 
+	settings := options{}
 	for _, opt := range opts {
-		if err := opt.Customize(&genericContainerReq); err != nil {
-			return nil, err
+		if opt, ok := opt.(Option); ok {
+			if err := opt(&settings); err != nil {
+				return nil, err
+			}
 		}
 	}
 
-	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
-	var c *RedisContainer
-	if container != nil {
-		c = &RedisContainer{Container: container}
-	}
+	tcOpts := []testcontainers.ContainerCustomizer{}
 
-	if err != nil {
-		return c, fmt.Errorf("generic container: %w", err)
+	waitStrategies := []wait.Strategy{
+		wait.ForListeningPort(redisPort).WithStartupTimeout(time.Second * 10),
+		wait.ForLog("* Ready to accept connections"),
```

Comment:
question: do we need the ForLog?

Reply:
I'd keep it for consistency with previous behaviour.

We could have an issue to revisit all modules and simplify the wait strategies removing the for log